### ISSUE TITLE
Feature Request - read and display Pokémon IVs

### ISSUE THREAD

**Issue by CharlyCheater** [AHAAAAAAA/PokemonGo-Map#2165]
As the IVs are the same for every trainer, wouldn't it be possible that the workers start the encounter with wild Pokémons and flee immediately but save the IVs of the encountered Pokémon? Would be pretty nice to have them displayed on the map.

**Comment by Sejadis** [comment]
i agree it would be nice but wouldnt that slow down the search extremely?

**Comment by CharlyCheater** [comment]
Yeah I guess one would need to use more workers with smaller steps each

**Comment by hgentry** [comment]
IVs are not determined until the battle starts; there's no way for the map to show this data.

**Comment by CharlyCheater** [comment]
[...] start the encounter with wild Pokémons and flee immediately [...]

**Comment by InflamedSebi** [comment]
kinda duplicate of https://github.com/AHAAAAAAA/PokemonGo-Map/issues/2080

**Comment by CharlyCheater** [comment]
Well, CP can be different for everyone but IVs not

**Comment by InflamedSebi** [comment]
well, I've been told a pokemon spawns with a CP value and gets scaled down by the trainer level.

nvm. You have to catch a pokemon to see IVs. And like explained there this is only possible by creating a fully automated farm bot, which shouldn't be made.

**Comment by CharlyCheater** [comment]
And that's where you are wrong.
The IVs are sent by the server as soon as you start the encounter. No need to actually catch the pokemon. You can flee immediately.

**Comment by RollTideDG** [comment]
Charly is right. I am reading IV's when I start an encounter with my MITM proxy.

**Comment by InflamedSebi** [comment]
oh, I apologize. Then I have to support this idea. Maybe like the current filter, so you can specify which pokemon to scan for their IVs and you don't scan every Weedle, Pidgey and Zubat for IVs.

**Comment by theblackmarmoset** [comment]
What is IV?

**Comment by RollTideDG** [comment]
IV is how perfect a pokemon is or % perfect. It is calculated by this formula:
(IndividualAttack + IndividualDefense + IndividualStamina) / 45 * 100

The 3 values you need are hidden by NIA but are sent in the requests. These are the DNA of your poke and never change for the life of it. Obviously a 100% perfect magikarp would be more desirable then say an 80% one even if they had the same CP. Knowing this value is critical when you are hunting for the best pokes

**Comment by theblackmarmoset** [comment]
Ahhh okay.  :)

**Comment by andikatanpah** [comment]
@RollTideDG hi bro, how to get The 3 values for IV?

**Comment by RollTideDG** [comment]
Right now the only way I know of is to use a MITM proxy and check your inventory or encounter a pokemon.
